apps/test/solutions/applab/ec_design.js

```diff
@@ -185,6 +241,65 @@ module.exports = {
     },
 
     {
+      description: "resizing button",
+      editCode: true,
+      xml: '',
+      runBeforeClick: function (assert) {
+        $("#designModeButton").click();
+        testUtils.dragToVisualization('BUTTON', 10, 20);
+        assertPropertyRowValue(0, 'id', 'button1', assert);
+        assertPropertyRowValue(2, 'width (px)', 80, assert);
+        assertPropertyRowValue(3, 'height (px)', 30, assert);
+        assertPropertyRowValue(4, 'x position (px)', 10, assert);
+        assertPropertyRowValue(5, 'y position (px)', 20, assert);
+
+        resizeElement(document.getElementById('design_button1'), 20, 0);
+        assertPropertyRowValue(2, 'width (px)', 100, assert);
+        assertPropertyRowValue(3, 'height (px)', 30, assert);
+
+        resizeElement(document.getElementById('design_button1'), 0, 20);
+        assertPropertyRowValue(2, 'width (px)', 100, assert);
+        assertPropertyRowValue(3, 'height (px)', 50, assert);
+
+        Applab.onPuzzleComplete();
+      },
+      expected: {
+        result: true,
+        testResult: TestResults.FREE_PLAY
+      },
+    },
+
+    {
+      description: "dragging button",
+      editCode: true,
+      xml: '',
+      runBeforeClick: function (assert) {
+        $("#designModeButton").click();
+        testUtils.dragToVisualization('BUTTON', 10, 20);
+        assertPropertyRowValue(0, 'id', 'button1', assert);
+        assertPropertyRowValue(2, 'width (px)', 80, assert);
+        assertPropertyRowValue(3, 'height (px)', 30, assert);
+        assertPropertyRowValue(4, 'x position (px)', 10, assert);
+        assertPropertyRowValue(5, 'y position (px)', 20, assert);
+
+        dragElement(document.getElementById('design_button1'), 20, 0);
+        assertPropertyRowValue(4, 'x position (px)', 20, assert);
+        assertPropertyRowValue(5, 'y position (px)', 20, assert);
+
+        dragElement(document.getElementById('design_button1'), 0, 20);
+        assertPropertyRowValue(4, 'x position (px)', 20, assert);
+        assertPropertyRowValue(5, 'y position (px)', 30, assert);
```

Comment:
It would be helpful to add a mousemove event at the "end" for drag as well, so you can get rid of the same weird 0.5 scale factor here.